src/main/resources/Diagram/DiagramViewSheet.xml

```diff
@@ -319,6 +325,26 @@ require.config({
     diagramViewerDeferred.reject();
   });
 
+  // Return a diagram in XML format if is any present in URL
+  function getDiagramXMLFromURL() {
+    if (window.location.hash.substring(0, 2) == '#R') {
+        // We need to exlude first 2 letters from hash (#R).
+        let decodedData = decodeURIComponent(window.location.hash.substring(2));
+        let graph = new Graph();
+        let rawDiagram = graph.decompress(decodedData);
+        return rawDiagram;
+    }
+    return null;
+  };
+
+  function getDiagramXML(container) {
+    let rawDiagram = getDiagramXMLFromURL();
+    if (null === rawDiagram) {
+      rawDiagram = container.data('model');
+    }
+    return rawDiagram;
```

Comment:
The previous 5 lines can also be written as:

    return getDiagramXMLFromURL() || container.data('model');